Source/common/SNTConfigurator.m

```diff
@@ -322,70 +354,85 @@ - (NSString *)machineID {
 }
 
 - (void)reloadConfigData {
-  NSFileManager *fm = [NSFileManager defaultManager];
-  if (![fm fileExistsAtPath:self.configFilePath]) {
-    // As soon as saveConfigToDisk is called, reloadConfigData will be called again because
-    // of the SNTFileWatchers on the config path. No need to use dictionaryWithCapacity: here.
-    self.configData = [NSMutableDictionary dictionary];
+  // Load the mobileconfig
+  self.configData = [self mobileConfig];
+  if (!self.configData[kClientModeKey]) {
+    // Default to Monitor if the config is missing or invalid
     self.configData[kClientModeKey] = @(SNTClientModeMonitor);
-    [self saveConfigToDisk];
-    return;
-  };
+  }
+
+  // Nothing else to do if a sync server is not involed
+  if (!self.configData[kSyncBaseURLKey]) return;
+
+  // Load the last known sync state
+  NSMutableDictionary *syncState =
+      [NSMutableDictionary dictionaryWithCapacity:[self syncServerKeys].count];
+  if (![[NSFileManager defaultManager] fileExistsAtPath:kSyncStateFilePath]) return;
```

Comment:
Just to be clear: if the syncStateFile doesn't exist, we don't want to call saveSyncStateToDisk?

Reply:
Yes, and actually we don't need to write to disk here at all. The sync-state should only be updated when a sync happens causing  santactl to send an XPC message to santad to update the file.